compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -213,7 +233,35 @@ pub struct StructType {
 }
 
 /// Corresponds to generic lists such as `<T, U>` in the source program.
-pub type Generics = Vec<TypeVariable>;
+/// Used mainly for resolved types which no longer need the information
+/// that is checked during resolution/type checking.
```

Comment:
This sounds a bit vague - what information? I think we could explicitly say "which no longer need names or kinds" or similar. Although I think the kinds would still be useful to have - I'm trying to think of a scenario where we may need them. We may need to check them when instantiating a function to ensure we're instantiating it with the correct kinds of types.

Reply:
I expanded the comment to include that this type is used when names or kinds are no longer needed. It should be trivial to switch `GenericTypeVars` out for `Generics` wherever `GenericTypeVars` are used. `GenericTypeVars` is really just a rename of the type that already pre-existed this PR as I wanted to avoiding changing the type in more places in this PR.